launcher/ui/pages/instance/StoragePage.cpp

```diff
@@ -26,27 +26,24 @@
 
 namespace fs = std::filesystem;
 
-unsigned get_directory_size(std::string path)
+unsigned getDirectorySize(QString path)
 {
-    if (!fs::exists(path))
-        return 0;
-    unsigned file_size_total = 0;
-    for (auto const& entry : fs::directory_iterator(path)) {
-        if (entry.is_directory())
-            file_size_total += get_directory_size(entry.path().string());
-        else
-            file_size_total += entry.file_size();
+    QDirIterator it(path, QDirIterator::Subdirectories);
+    qint64 fileSizeTotal = 0;
+    while (it.hasNext()) {
+        fileSizeTotal += it.fileInfo().size();
+        it.next();
     }
-    return file_size_total;
+    return fileSizeTotal;
 }
 
-void clear_directory_inner(std::string path)
+void clearDirectoryInner(QString path)
 {
-    if (!fs::exists(path))
-        return;
-
-    for (auto const& entry : fs::directory_iterator(path))
-        fs::remove_all(entry);
+    QDirIterator it(path, QDirIterator::Subdirectories);
+    while (it.hasNext()) {
+        fs::remove_all(it.path().toStdString());
+        it.next();
+    }
```

Comment:
I believe this can be replaced with `QDir(path).removeRecursively()`.
It might also be a good idea to store the QDir as a variable then display a message box notifying the user that an error occurred during the removal should removeRecursively return false.